webapp/sources/rudder/rudder-web/src/main/elm/sources/NodeCompliance/ApiCalls.elm

```diff
@@ -31,6 +32,38 @@ getPolicyMode model =
   in
     req
 
+getRulesInfo : Model -> Cmd Msg
+getRulesInfo model =
+  let
+    req =
+      request
+        { method  = "GET"
+        , headers = []
+        , url     = getUrl model [ "rules" ] []
+        , body    = emptyBody
+        , expect  = expectJson GetRulesInfo decodeGetRules
+        , timeout = Nothing
+        , tracker = Nothing
+        }
+  in
+    req
+
+getDirectivesInfo : Model -> Cmd Msg
+getDirectivesInfo model =
+  let
+    req =
+      request
+        { method  = "GET"
+        , headers = []
+        , url     = getUrl model [ "directives" ] []
+        , body    = emptyBody
+        , expect  = expectJson GetDirectivesInfo decodeGetDirectives
+        , timeout = Nothing
+        , tracker = Nothing
+        }
+  in
+    req
+
```

Comment:
I don't think we want to do that. It will be extremely costly when the number of rules/directives growths, especially since we get all rules/directives, and not the ones for the node. 

A better approach would be to extends `getNodeComplianceId` api to provide the compliance mode - and moreover, I'm pretty sure it's something that would be interesting to have in the public API too.

Reply:
The API part is done here : https://github.com/Normation/rudder/pull/5308/files
You can change the decoder at the level of the `RuleCompliance` and `DirectiveCompliance` data types : just adding a `policyMode` field will do, it simplifies also a lot of things in this PR I guess :smile: